DependencyInjection/CompilerPass/PluginPass.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+namespace Http\HttplugBundle\DependencyInjection\CompilerPass;
+
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+
+/**
+ * Remove plugins if their dependencies are missing.
+ *
+ * @author Tobias Nyholm <tobias.nyholm@gmail.com>
+ */
+class PluginPass implements CompilerPassInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function process(ContainerBuilder $container)
+    {
+        if (!($container->hasDefinition('logger') || $container->hasAlias('logger'))) {
```

Comment:
I would like to avoid container parameters if we could. 

We could fetch the `httplug.plugin.logger` definition and look at it's dependencies and validate if they are equal to the default value and then check if they exists. But that does not solve the issue you brought up in https://github.com/php-http/HttplugBundle/pull/37/files#r50537858

Hm, we maybe need to use container parameters anyway.